c/src/v4/signing.c

```diff
@@ -244,3 +245,143 @@ int sbp_msg_ed25519_certificate_cmp(const sbp_msg_ed25519_certificate_t *a,
   }
   return ret;
 }
+
+bool sbp_msg_ed25519_signature_encode_internal(
+    sbp_encode_ctx_t *ctx, const sbp_msg_ed25519_signature_t *msg) {
+  if (!sbp_u8_encode(ctx, &msg->stream_counter)) {
+    return false;
+  }
+  if (!sbp_u8_encode(ctx, &msg->on_demand_counter)) {
+    return false;
+  }
```

Comment:
Requirement says: 

"The proposal is to have two separate counters.  One to count the messages in the stream and a separate counter for burst or on-demand messages. "

It's not clear to me whether having both (being one static) is desired for this feature.

Reply:
I think the situation in which you'd use both is - you have "normal" groups coming in response to a CRA. via the counter you know you missed one, you send a new CRA message with the correction mask set so that you can request whatever you missed, at this point you'd be getting both "normal" and on demand groups

Reply:
Not entirely sure if that's how it'll work though. Like if you send two CRAs like that, would the second one "cancel" the first one? If so then one counter would be enough I think

Reply:
Agreed